src/backend/opencl/kernel/iops.cl

```diff
@@ -23,20 +31,22 @@ void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 #ifdef MAX_OP
 void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 {
-    if ((sabs(lhs[0]) < sabs(rhs)) ||
-        (sabs(lhs[0]) == sabs(rhs) && *lidx > ridx)) {
+    if (((sabs(lhs[0]) < sabs(rhs)) ||
+         (sabs(lhs[0]) == sabs(rhs) &&
+          *lidx > ridx)) &&
+        !is_nan(lhs[0])) {
         *lhs = rhs;
         *lidx = ridx;
     }
 }
 #endif
 #else
-#define sabs(in) in
 #ifdef MIN_OP
 void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 {
-    if ((*lhs > rhs) ||
-        (*lhs == rhs && *lidx < ridx)) {
+    if (((*lhs > rhs) ||
+         (*lhs == rhs && *lidx < ridx)) &&
+        !is_nan(*lhs)) {
```

Comment:
Is rhs guaranteed to be a non-nan number? Do we need to add a check for that as well?

Reply:
Actually... the way this was originally written kinda made me confused because the order of lhs and rhs is flipped on the other backends. In any case though, I think I should be checking if `rhs` is nan here, instead of `*lhs`. `*lhs` (pointing to the kernel's `out_val`) is actually guaranteed to be non-nan, since it's initialized to the type's reduction initial value in the beginning (in `ireduce_first_kernel()`), and there is a guard there that prevents it from being changed to an element of the input array if that element is nan. The same guard then should be applied in the operator (in this case checking if `rhs` is nan) to maintain that guarantee.